front/lib/resources/data_source_resource.ts

```diff
@@ -107,14 +107,29 @@ export class DataSourceResource extends ResourceWithVault<DataSource> {
     name: string,
     options?: Omit<FetchDataSourceOptions, "limit" | "order">
   ): Promise<DataSourceResource | null> {
-    const [dataSource] = await this.baseFetchWithAuthorization(auth, {
+    const dataSources = await this.fetchByNames(auth, [name], options);
+    if (dataSources.length === 0) {
```

Comment:
raise if > 1 ?

Reply:
It can't happen as we have a unique index on `workspaceId, name`.

https://github.com/dust-tt/dust/blob/deb4a2465797768f8dfcc04b8be77677a82df645/front/lib/models/data_source.ts#L97